articles/most-useful-nodejs-packages/index.md

```diff
@@ -0,0 +1,424 @@
+-   [`Mocha`](https://www.npmjs.com/package/mocha) - helps you to carry out asynchronous tests.
+-   [`Chai`](https://www.chaijs.com/) - is a BDD/TDD assertion library for Node.js and browsers that can be paired with any JavaScript testing framework.
+-   [`Jest`](https://jestjs.io/) - JavaScript testing framework which focuses on simplicity.
+
+### 10. MySQL
+[MySQL](https://en.wikipedia.org/wiki/MySQL) is a Node.js client for the MySQL protocol. Before using MySQL to connect to your database, ensure you have MySQL installed and configured in your machine. Then create a database and database table that you can work with.
+
+**An Example on How to Use MySQL with Node.js**
+
+```js
+//use require() to include MySql library
+var mysqldb = require('mysql');
+//you need to create a connection to the database
+//make sure you replace 'user' and 'password' with your correct values
+//create connection
+var connection = mysqldb.createConnection({
+    host: 'localhost',
+    user: 'you',
+    password: 'passwordgoeshere',
+    database: 'mydb'
+});
+connection.connect();
+connection.query('SELECT 1 + 1 AS solution', function(error, results, fields) {
+    if (error) throw error;
+    console.log('The solution is: ', results[0].solution);
+});
+connection.end();
+//terminate connection
+//executes the remaining queries
+//send quit packet to your MySql server
+//end
+```
+[Code snippet source](https://github.com/mysqljs/mysql).
+
+From the example above:
+-   Every method that you invoke on a database connection is queued and executed in sequence.
+-   To terminate/close the connection use `end()`, this makes sure that all the remaining queries are executed before sending a quit packet to the MySQL server.
+
+#### Other MySQL and Helper Frameworks Include:
+-   [`Sequelize`](https://www.npmjs.com/package/sequelize) - a promised based Node.js Object-Relation Mapping for MySQL, SQLite, Maria DB, Postgres, and Microsoft SQL Server.
+-   [`Knex.js`](https://knexjs.org/) - is designed to be flexible, portable and fun to run for MSSAL, MySQL, Oracle, MariaDB, SQLite3 and Amazon Redshift.
+
+#### Some Mongo database and Helper Frameworks Include
+-   [`Mongoose`](https://www.npmjs.com/package/mongoose) - is a [MongoDB](https://www.mongodb.com/) object modeling tool for asynchronous environments.
+-   [`Mongoose Paginate v2`](https://www.npmjs.com/package/mongoose-paginate-v2) - used for pagination query for Mongoose.
+
+### 11. Nodemon
+[Nodemon](https://www.npmjs.com/package/nodemon) is a monitoring tool and it helps Node.js developers by automatically restart an application when file changes in the app directory are detected. With Nodemon, you do not need any additional code or development method changes. It is a replacement wrapper for Node.js.
+
+Nodemon is simple to use. You just need to replace the word node with `nodemon` on the command-line when executing your `script`, i.e. `nodemon ./server.js localhost 300`
```

Comment:
Consider highlighting the fact that users have to install Nodemon globally `-g` for the nodemon command to work. A local installation uses `npx nodemon` instead.